src/python/pants/engine/README.md

```diff
@@ -181,3 +225,108 @@ in the context of scala and mixed scala & java builds.  Twitter spiked on a proj
 a target-level scheduling system scoped to just the jvm compilation tasks.  This bore fruit and
 served as further impetus to get a "tuple-engine" designed and constructed to bring the benefits
 seen in the jvm compilers to the wider pants world of tasks.
+
+## Datatypes in Depth
```

Comment:
This section... feels like too much. But maybe it's not obvious to me yet that people will need to be convinced to add types to things... in the presence of `@rules`? The idea behind `@rules` is really to force people to use types, whether or not we motivate them to in some other way.

Maybe if you trimmed it down a bit?

Reply:
That makes sense to me. Would it be better just removed?

Reply:
I think that incorporating a small amount of "types are good, use them!" into the section that already describes `datatype` would be good? But probably not much more than a paragraph.

Sorry... it's great writing, but it doesn't feel like the right place.

Reply:
Made that edit in af466ad.